index.bs

```diff
@@ -644,6 +639,31 @@ The [=reading change threshold|threshold=] value depends on the surrounding soft
 environment constraints, e.g., software power consumption optimizations or the underlying
 [=device sensor=]'s accuracy.
 
+## Sampling Frequency and Reporting Frequency ## {#concepts-sampling-and-reporting-frequencies}
+
+For the purpose of this specification, <dfn>sampling frequency</dfn> for a [=platform sensor=] is
+defined as a frequency at which the user agent obtains [=sensor readings=] from the underlying
+platform.
+
+The user agent can request the underlying platform to deliver [=sensor readings|readings=] at a certain
+rate which is called <dfn>requested sampling frequency</dfn>.
+
+The [=sampling frequency=] is equal to the [=requested sampling frequency=] if the underlying platform
+can support it.
+
+The [=sampling frequency=] differs from the [=requested sampling frequency=] in the following cases:
+ - the [=requested sampling frequency=] exceeds upper or lower [=sampling frequency=] bounds
+   supported by the underlying platform.
+ - the [=reading change threshold|threshold=] value is significant so that some of the
+   [=device sensor=]' measurements are skipped and the [=sensor readings=] are not updated.
+
+The <dfn>reporting frequency</dfn> for a concrete {{Sensor}} object is defined as a frequency at which
+{{Sensor/onreading}} notification is called.
+
+Whereas a {{Sensor}} object cannot access new [=sensor readings|readings=] at a higher rate than the
```

Comment:
nit: Whereas ... cannot ..., the reporting frequency can never exceed.

Usually, `whereas`  used to represent contrast, in this case, both parts of sentence is 'cannot', sounds bit strange.

Reply:
done